Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py

```diff
@@ -0,0 +1,316 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+''' IMPORTS '''
+
+import json
+import requests
+from requests import Request, Session
+
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+USERNAME = demisto.params().get('credentials').get('identifier')
+PASSWORD = demisto.params().get('credentials').get('password')
+API_KEY = demisto.params().get('key')
+# Remove trailing slash to prevent wrong URL path to service
+SERVER = demisto.params()['url'][:-1] \
+    if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+# Should we use SSL
+USE_SSL = not demisto.params().get('insecure', False)
+# Service base URL
+BASE_URL = SERVER + '/BeyondTrust/api/public/v3'
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': 'PS-Auth key=' + API_KEY + '; ' + 'runas=' + USERNAME + '; ' + 'pwd=[' + PASSWORD + '];'
+}
+
+SESSION = requests.session()
+# Remove proxy if not set to true in params
+if not demisto.params().get('proxy'):
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
+
+
+''' HELPER FUNCTIONS '''
+
+
+def http_request(method, suffix_url, params=None, data=None):
+
+    url = BASE_URL + suffix_url
+
+    res = SESSION.request(
+        method,
+        url,
+        verify=USE_SSL,
+        params=params,
+        data=data,
+        headers=HEADERS
+    )
+    # Handle error responses gracefully
+    if res.status_code not in {200, 201, 204}:
```

Comment:
are all these status codes needed ?

Reply:
yes, I have a few commands that return 201 or 204 as a Success status code